pyads/get_bad_active_creatives.py

```diff
@@ -0,0 +1,177 @@
+    ))
+    if 'results' in response and len(response['results']):
+      for line_item in response['results']:
+        db_line_items[str(line_item['id'])] = line_item
+        if line_item['endDateTime'] is not None:
+          date = line_item['endDateTime']['date']
+          if datetime(date['year'], date['month'], date['day']) > datetime.now():
+            active_line_items[line_item['id']] = line_item['name']
+        else:
+          active_line_items[line_item['id']] = line_item['name']
+      statement.offset += statement.limit
+    else:
+      break
+
+  db_line_items['active_line_items'] = active_line_items
+  db_line_items.close()
+
+  print("LINE ITEMS FETCHED: {}".format(len(active_line_items)))
+
+
+def fetch_associatiated_creatives(client):
+
+  creatives = {}
+  db_creatives = shelve.open('data/creatives.db')
+
+  lica_service = client.GetService(
+    'LineItemCreativeAssociationService', version='v201805')
+
+  # Get All creatives associated with active line items
+  print('GET CREATIVES FROM LINE ITEMS')
+
+  for line_item_id in tqdm(active_line_items):
+    statement = (ad_manager.StatementBuilder())
+
+    while True:
+      response = lica_service.getLineItemCreativeAssociationsByStatement(
+        statement.ToStatement())
+      if 'results' in response and len(response['results']):
+        for lica in response['results']:
+          if lica['status'] == "ACTIVE":
+            creatives[lica['creativeId']] = lica['lineItemId']
+            db_creatives[str(lica['creativeId'])] = lica['lineItemId']
+        statement.offset += statement.limit
+      else:
+        break
+
+  db_creatives['creatives'] = creatives
+  db_creatives.close()
+
+  print('Numer of associations fetched: {}'.format(len(creatives)))
+
+
+def fetch_creatives(client, creatives_to_scan):
+
+  db_creatives_to_scan = shelve.open('data/creatives_to_scan.db')
+
+  creative_service = client.GetService(
+    'CreativeService', version='v201805')
+
+  # Look for text in all creatives
+  print('SCANNING CREATIVES')
+
+  for creative_id in tqdm(creatives_to_scan):
+    statement = (ad_manager.StatementBuilder()
+                 .Where('creativeId = :creativeId')
+                 .WithBindVariable('creativeId', creative_id))
+
+    while True:
+      response = creative_service.getCreativesByStatement(statement.ToStatement())
+      if 'results' in response and len(response['results']):
+        for creative in response['results']:
+          db_creatives_to_scan[str(creative['id'])] = creative
+        statement.offset += statement.limit
+      else:
+        break
+
+  print('Numer of creatives feteched: {}'.format(len(db_creatives_to_scan.items())))
+
+
+def main(client):
+  parser = argparse.ArgumentParser(description='--fetch if you want to fetch dfp configs')
+  parser.add_argument('--fetch', help='use if you want to fetch data', action="store_true")
+  args = parser.parse_args()
+
+  text_to_look_for = "http://"
```

Comment:
as this is the most crucial var in this script, I would move it to the top of the file and rename it to `TEXT_TO_LOOK_FOR`